src/views/RecIM/components/Forms/ActivityForm/index.jsx

```diff
@@ -309,30 +349,122 @@ const ActivityForm = ({
     setNewInfo(currentInfo);
+    setCropperImageData(null);
   };
 
+  /**********************************************************
+  /*Following functions are solely related to photo submission*
+  /**********************************************************/
+
+  async function clearPhotoDialogErrorTimeout() {
+    clearTimeout(photoDialogErrorTimeout);
+    setPhotoDialogErrorTimeout(null);
+    setPhotoDialogError(null);
+  }
+
+  /**
+   * Creates the Photo Dialog message that will be displayed to the user
+   *
+   * @returns {string} The message of the Photo Dialog
+   */
+  function createPhotoDialogBoxMessage() {
+    let message = '';
+    // If an error occured and there's no currently running timeout, the error is displayed
+    // and a timeout for that error message is created
+    if (photoDialogError !== null) {
+      message = <span style={{ color: '#B63228' }}>{photoDialogError}</span>;
+      if (photoDialogErrorTimeout === null) {
+        // Shows the error message for 6 seconds and then returns back to normal text
+        setPhotoDialogErrorTimeout(
+          setTimeout(() => {
+            setPhotoDialogErrorTimeout(null);
+            setPhotoDialogError(null);
+          }, 6000),
+        );
+      }
+    }
+
+    // If no error occured and the cropper is shown, the cropper text is displayed
+    else if (cropperImageData) {
+      message = 'Drag & Drop Picture, or Click to Browse Files';
+    }
+
+    // If no error occured and the cropper is not shown, the pick a file text is displayed
+    else {
+      message = isMobile
+        ? 'Tap Image to Browse Files'
+        : 'Drag & Drop Picture, or Click to Browse Files';
+    }
+    return message;
+  }
+
+  function onCropperZoom(event) {
+    if (event.detail.ratio > 1) {
+      event.preventDefault();
+      cropperRef.current.cropper.zoomTo(1);
+    }
+  }
+
+  /**
+   * Handles the acceptance of the user dropping an image in the Photo Uploader in News submission
+   *
+   * @param {*} fileList The image dropped in the Dropzone of the Photo Uploader
+   */
+  function onDropAccepted(fileList) {
+    var previewImageFile = fileList[0];
+    var reader = new FileReader();
+    reader.onload = () => {
+      imageOnLoadHelper(reader);
+    };
+    reader.readAsDataURL(previewImageFile);
+  }
+
+  /**
+   * Handles the rejection of the user dropping an invalid file in the Photo Updater Dialog Box
+   * Copied from Identification
+   */
+  async function onDropRejected() {
+    await clearPhotoDialogErrorTimeout();
+    setPhotoDialogError('Sorry, invalid image file! Only PNG and JPEG images are accepted.');
+  }
+
+  function imageOnLoadHelper(reader) {
+    var dataURL = reader.result.toString();
+    var i = new Image();
+    i.onload = async () => {
+      var aRatio = i.width / i.height;
+      setAspectRatio(aRatio);
+      setPhotoDialogError(null);
+      setAspectRatio(aRatio);
+      setCropperImageData(dataURL);
+    };
+    i.src = dataURL;
+  }
+
```

Comment:
We have a `RecIM/components/Helpers` file that this should be stored in. With the current refactor in progress, it would be beneficial to not add this workload in parallel to current refactor.